Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -0,0 +1,1094 @@
+//
+//  ReplicatorCollectionSGTest.cc
+//
+//  Copyright 2022-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+//
+
+#include "c4Base.h"
+#include "ReplicatorAPITest.hh"
+#include "ReplicatorOptions.hh"
+#include "ReplicatorLoopbackTest.hh"
+#include "Base64.hh"
+#include "Defer.hh"
+#include "c4Collection.h"
+#include "c4DocEnumerator.h"
+#include "c4Document.h"
+#include "c4Database.h"
+#include "c4Replicator.hh"
+#include "fleece/Mutable.hh"
+#include <array>
+
+// Tests in this file, tagged by [.SyncServerCollection], are not done automatically in the
+// Jenkins/GitHub CI. They can be run in locally with the following environment.
+// Couchbase DB server, with docker, for example,
+//   docker run -d --name cbserver -p 8091-8096:8091-8096 -p 11210-11211:11210-11211 couchbase:7.1.1
+//   bucket configuration:
+//     user    : Administrator
+//     password: password
+//     name    : any
+//     scope   : flowers
+//     collection: roses
+// Sync-gateway:
+//   config.json:
+/*
+ {
+   "bootstrap": {
+     "server": "couchbase://localhost",
+     "username": "Administrator",
+     "password": "password",
+     "use_tls_server": false
+   },
+   "logging": {
+     "console": {
+       "log_level": "info",
+       "log_keys": ["*"]
+     }
+   }
+ }
+ */
+//  config db:
+/*
+ curl --location --request PUT "localhost:4985/scratch/" \
+ --header "Content-Type: application/json" \
+ --header "Authorization: Basic QWRtaW5pc3RyYXRvcjpwYXNzd29yZA==" \
+ --data-raw '{"num_index_replicas": 0, "bucket": "your_bucket_name", "scopes": {"flowers": {"collections":{"flowers":{}}}}}'
+ */
+//  config SG user:
+/*
+ curl --location --request POST "localhost:4985/scratch/_user/" \
+ --header "Content-Type: application/json" \
+ --header "Authorization: Basic QWRtaW5pc3RyYXRvcjpwYXNzd29yZA==" \
+ --data-raw '{"name": "sguser", "password": "password", "admin_channels": ["*"]}'
+ */
```

Comment:
* Would it be possible to create a static SG config file to configure SG? Previous test setup has two config files : TLS and Non-TLS.

* Would it be possible to have a script that helps running the docker and also configure the buckets?

Reply:
1. I got the impression that static SG config is obsolete, contraindicated
2. yes. we may provide the script for Unix shell.

Reply:
Here is a bit more info from Adam:

```
It's correct to say that we won't support file-based database configuration forever, and the recommended
approach for 3.0 and later is to use file-based config only for cluster bootstrapping, and to use the REST API
for database and user creation/management
```

Reply:
As the config for tests are pretty much static, we could move the CURL commands to the same script that starts SG. If there are some changes in the test code, the test code will need to do the REST API call.

Reply:
I haven't checked in any script yet. I hope we will script to do the SG and CB server. Jim has script in his docker compose. I just did not get to it and manually set up the cb server. I hope not to include it in this PR if okay.